docs/adding_provider_support.md

```diff
@@ -1,71 +1,168 @@
+external CCM (if applicable) are obtained. First, you need to add a constant
+with the environment variable name used by the `machine-controller` to the
+[`pkg/util/credentials`][12] package. Then, you need to add the logic for
+obtaining credentials to the [`ProviderCredentials` function][13]. Credentials
+are usually obtained by reading them from the environment variables, but custom
+workflows are supported as well. For example AWS uses the AWS SDK to fetch the
+credentials.
+
+With the API types added to support the new provider, you need to implement
+Terraform integration for worker nodes.
 
 ## Implementing Terraform Integration for Worker Nodes
 
-The Terraform Integration for worker nodes is used to source common information used to create the worker nodes. For example, information such as region, instance size, VPC IDs, are usually same as for the control plane nodes. If infrastructure is provisioned using Terraform, those information can be sourced from the Terraform state, so operator doesn't have to manually provide them upon provisioning the cluster. The Terraform integration is implemented in the [`pkg/terraform` package](https://github.com/kubermatic/kubeone/tree/ec8bf305446ac22529e9683fd4ce3c9abf753d1e/pkg/terraform).
+The Terraform Integration for worker nodes is used to source common information
+used to create the worker nodes. For example, information such as region,
+instance size, VPC IDs, are usually same as for the control plane nodes. If
+infrastructure is provisioned using Terraform, those information can be sourced
+from the Terraform state, so operator doesn't have to manually provide them upon
+provisioning the cluster. The Terraform integration is implemented in the
+[`pkg/terraform` package][14].
 
-For the reference what fields are part of the WorkerSet, check the `CloudProviderSpec` API implemented in the machine-controller. You can find example manifests with all available options in the [`examples` directory in the `machine-controller` repository](https://github.com/kubermatic/machine-controller/tree/master/examples) or check out the [API implementation](https://github.com/kubermatic/machine-controller/tree/master/pkg/cloudprovider/provider) for more details.
+For the reference what fields are part of the WorkerSet, check the
+`CloudProviderSpec` API implemented in the machine-controller. You can find
+example manifests with all available options in the `examples` directory in the
+[`machine-controller` repository][15] or check out the [API implementation][16]
+for more details.
 
-First, you need to implement a `WorkerConfig` structure in the `pkg/terraform` package with fields that will be sourced from the Terraform state. You can refer to the [`awsWorkerConfig` structure](https://github.com/kubermatic/kubeone/blob/ec8bf305446ac22529e9683fd4ce3c9abf753d1e/pkg/terraform/config.go#L39-L49) as an example.
+First, you need to implement a `WorkerConfig` structure in the `pkg/terraform`
```

Comment:
Those structures aren't in the `pkg/terraform` package anymore.

Reply:
nice catch 👍 fixed